main.go

```diff
@@ -16,17 +19,51 @@ func proxyHandler(w http.ResponseWriter, r *http.Request) {
 		"header":     r.Header,
 	}).Info("Received /proxy request")
 
+	target := r.Header.Get("X-OXYproxy-target")
+	if target == "" {
+		log.Warning("Target source is undefined")
```

Comment:
ok, you added a log. But probably its better solution to return http error to the user? 
or how do you plan to continue proxing flow without target?

Reply:
solved. Logger tells the server about the nil target and then we send response status to client.

Reply:
1) you forgot to return in the error case
2) semantically "bad request" status is more suitable here